src/drivers/hmc5883/hmc5883.cpp

```diff
@@ -1017,10 +1020,17 @@ HMC5883::collect()
 	rotate_3f(_rotation, xraw_f, yraw_f, zraw_f);
 
 	new_report.x = ((xraw_f * _range_scale) - _scale.x_offset) * _scale.x_scale;
+	new_report.x += ((yraw_f * _range_scale) - _scale.y_offset) * _scale.x_offdiag;
+	new_report.x += ((zraw_f * _range_scale) - _scale.z_offset) * _scale.y_offdiag;
+
 	/* flip axes and negate value for y */
 	new_report.y = ((yraw_f * _range_scale) - _scale.y_offset) * _scale.y_scale;
+	new_report.y += ((xraw_f * _range_scale) - _scale.x_offset) * _scale.x_offdiag;
+	new_report.z += ((zraw_f * _range_scale) - _scale.z_offset) * _scale.z_offdiag;
```

Comment:
This should be `new_report.y  += ((zraw_f * _range_scale) - _scale.z_offset) * _scale.z_offdiag;`